gobblin-compaction/src/main/java/gobblin/compaction/conditions/RecompactionConditionBasedOnFileCount.java

```diff
@@ -0,0 +1,25 @@
+package gobblin.compaction.conditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.compaction.dataset.DatasetMetric;
+
+/**
+ * An implementation {@link RecompactionCondition} which examines the number of files in the late outputDir
+ * If the file count exceeds the file count limit, a recompaction flow is triggered.
+ */
+public class RecompactionConditionBasedOnFileCount implements RecompactionCondition {
+
+  private int fileCountLimit = 0;
```

Comment:
All members should be be final unless they are not initialized in the constructor